### PR TITLE
Changes to enable Apache Phoenix tracing

### PR DESCRIPTION
**Pull request by bijugs**
The changes in this PR will help users trace [Phoenix](https://phoenix.apache.org/) queries from clients and collect statistics. This is based on Apache Phoenix documentation on [tracing](https://phoenix.apache.org/tracing.html).

### CODE REVIEW

**Comment by bijugs** [comment]
*Testing*
- set `` default["bcpc"]["hadoop"]["phoenix"]["tracing"]["enabled"] = `` to ``true`` in ``hbase.rb`` of ``bcpc-hadoop`` cookbook
- upload ``bcpc-hadoop`` cookbook
- run ``chef-client`` on all nodes running ``hbase`` processes (master & region server)
- Verify ``TRACING_STATS`` is created

**Comment by pu239ppy** [comment]
Built and verified testing steps

File: cookbooks/bcpc-hadoop/recipes/hbase_master.rb [hunk 1]
```diff
@@ -65,3 +66,20 @@
   subscribes :restart, "user_ulimit[hbase]", :delayed
   subscribes :restart, "template[/etc/hadoop/conf/core-site.xml]", :delayed
 end
+
+if node["bcpc"]["hadoop"]["phoenix"]["tracing"]["enabled"]
+
+  template "/tmp/trace_table.sql" do
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/recipes/hbase_master.rb]: This should use the Chef cache and not `/tmp`

File: cookbooks/bcpc-hadoop/recipes/hbase_master.rb [hunk 2]
```diff
@@ -65,3 +66,20 @@
...
+    source "phoenix_trace-table.sql.erb"
+    mode "0755"
+    action :create
+  end
+
+  bash "create_phoenix_trace_table" do
+    code <<-EOH
+    HBASE_CONF_PATH=/etc/hadoop/conf:/etc/hbase/conf /usr/hdp/current/phoenix-client/bin/sqlline.py "#{node[:bcpc][:hadoop][:zookeeper][:servers].map{ |s| float_host(s[:hostname])}.join(",")}:#{node[:bcpc][:hadoop][:zookeeper][:port]}:/hbase" /tmp/trace_table.sql
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/recipes/hbase_master.rb]: This is a security issue as there is a race above I can have fun creating `/tmp/trace_table.sql` and now you are executing my arbitrary SQL as HBase; wahoo! (Please use the Chef cache.)

**Pull request merged**